src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,109 @@
+                try {
+                    int index = Integer.parseInt(argument) - 1;
+                    return new MarkAsDoneCommand(index);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("\u2639 OOPS!!! The index of a task must be a number.");
+                }
+            case UNMARK:
+                try {
+                    int index = Integer.parseInt(argument) - 1;
+                    return new UnmarkAsDoneCommand(index);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("\u2639 OOPS!!! The index of a task must be a number.");
+                }
+            case DELETE:
+                try {
+                    int index = Integer.parseInt(argument) - 1;
+                    return new DeleteCommand(index);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("\u2639 OOPS!!! The index of a task must be a number.");
+                }
+            case BYE:
+                return new ExitCommand();
+            case TODO: {
+                String description = argument;
+                Task task = new Todo(description);
+                return new AddCommand(task);
+            }
+            case DEADLINE:
+                try {
+                    inputs = argument.split(" /", 2);
+                    String description = inputs[0];
+                    String by = inputs[1].split(" ", 2)[1];
+                    Task task = new Deadline(description, by);
+                    return new AddCommand(task);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException("\u2639 OOPS!!! The deadline of a deadline cannot be empty.");
+                } catch (DateTimeParseException e) {
+                    throw new DukeException(
+                        "\u2639 OOPS!!! The date time format must be something like 2007-12-03T10:15:30.");
+                }
+            case EVENT:
+                try {
+                    inputs = argument.split(" /", 3);
+                    String description = inputs[0];
+                    String from = inputs[1].split(" ", 2)[1];
+                    String to = inputs[2].split(" ", 2)[1];
+                    Task task = new Event(description, from, to);
+                    return new AddCommand(task);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException(
+                        "\u2639 OOPS!!! The start time or end time of an event cannot be empty.");
+                } catch (DateTimeParseException e) {
+                    throw new DukeException(
+                        "\u2639 OOPS!!! The date time format must be something like 2007-12-03T10:15:30.");
+                }
+            case FIND:
+                return new FindCommand(argument);
+            default:
+                return new Command();
+            }
```

Comment:
Consider splitting this `switch` statement into multiple helper methods for easier readability.

For reference, see [Avoid deep nesting](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)